Library/Homebrew/cmd/update-bash.sh

```diff
@@ -2,22 +2,38 @@ brew() {
   "$HOMEBREW_BREW_FILE" "$@"
 }
 
+git() {
+  "$HOMEBREW_GIT" "$@"
+}
+
 which_git() {
-  local which_git
   local active_developer_dir
 
-  which_git="$(which git 2>/dev/null)"
-  if [[ -n "$which_git" && "/usr/bin/git" = "$which_git" ]]
+  if [[ -n "$HOMEBREW_GIT" && -x "$HOMEBREW_GIT" ]]
+  then
+    echo "$HOMEBREW_GIT"
+    return
+  fi
+
+  if [[ -n "$GIT" && -x "$GIT" ]]
+  then
+    HOMEBREW_GIT="$GIT"
+    echo "$HOMEBREW_GIT"
+    return
+  fi
+
+  HOMEBREW_GIT="$(which git 2>/dev/null)"
```

Comment:
> > I think it would be nice to maybe have some of this as a set environment variable in bin/brew.

> :+1: If we can do that in a way that doesn't slow down commands that don't require git at all.

I am not sure this is a good idea because not all of commands require `git` at all. Even more, for commands which require `git`, the ruby version of `Utils.ensure_git_installed!` and `Utils.git_available?` are still required. Setting the environment can be confusing in such circumstance.

How about creating a `Library/Utils/git.sh` file which is a bash equivalent of `Library/Homebrew/utils/git.rb` and only source it when needed for bash commands.